core/src/main/scala/org/apache/spark/deploy/security/DefaultHadoopAccessManager.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.security
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.FileSystem
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.security.UserGroupInformation
+
+class DefaultHadoopAccessManager(hadoopConf: Configuration) extends HadoopAccessManager {
+
+  def getTokenRenewer: String = {
+    UserGroupInformation.getCurrentUser.getShortUserName
```

Comment:
In general, do we want to depend on UGI for spark security ?
I would expect Kubernetes or spark standalone or some future resource manager to now be necessarily compatible with it (I was actually surprised the mesos is).
+CC @rxin

Can we externalize use of UGI from core ?

Yes, it has leaked into core in a few places - which is unfortunate IMO.